From f0dcda30d0148f10bb1de2bb70a88140d565d377 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 23 Apr 2016 12:26:40 +0200
Subject: [PATCH] [card] Use pa_card_put for PulseAudio 8.0

In PulseAudio 8.0, profile selection has been moved to pa_card_put, which now needs to be called in addition to pa_card_new.
Also, in PulseAudio 10.0, the pa_card_choose_initial_profile now needs to be done separately to properly choose the initial active profile.

See https://patchwork.freedesktop.org/patch/62759/ for details.
---
 src/droid/module-droid-card.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/droid/module-droid-card.c b/src/droid/module-droid-card.c
index 0b8c93f..98c4f0d 100644
--- a/src/droid/module-droid-card.c
+++ b/src/droid/module-droid-card.c
@@ -903,6 +903,13 @@ int pa__init(pa_module *m) {
     u->card->userdata = u;
     u->card->set_profile = card_set_profile;
 
+#if (PULSEAUDIO_VERSION >= 10)
+    pa_card_choose_initial_profile(u->card);
+#endif
+#if (PULSEAUDIO_VERSION >= 8)
+    pa_card_put(u->card);
+#endif
+
     u->modargs = ma;
     u->module = m;
 
